server/src/test/java/io/druid/server/coordinator/DruidCoordinatorTest.java

```diff
@@ -338,6 +338,10 @@ public void childEvent(
     Assert.assertEquals(1, segmentAvailability.size());
     Assert.assertEquals(0l, segmentAvailability.get(dataSource));
 
+    while (coordinator.getPendingSegments().get(dataSource).get() > 0) {
```

Comment:
Timeout is easy but implementing this by fully-event-driven-notify-system is rather a big work. Is it necessary just for fixing the test or is it a part of bigger plan?